### PR TITLE
feat: build new feed experts as a new agent and add the transfer widget

### PR DESCRIPTION
**Pull request by FrankLi123**
This PR focuses on the following tasks:

1. Adding an **Embedded Widget** for `transfer_expert` that provides a smooth transaction process based on user prompts.
   - Developed a new React component using `wagmi` for blockchain interactions and `rainbowkit` for wallet support.

2. Created a new agent file `feed_explorer.py` that has 2 new experts:

- `feed_source_expert` - retrieves activities of an account address based on data sources such as networks and platforms.

   - This helps users track activities on specific social platforms like `Farcaster`. Requests are made through the DSL Python SDK.

- `defi_expert` - monitors users' DeFi activities such as swaps and staking. Requests are made through the DSL Python SDK.

3. Create Unit Test Cases for `feed_explorer_agent`

### CODE REVIEW

File: src/tests/agent_trajectory/feed_explore.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        async def async_test():
+            events = feed_explorer_agent.astream_events({"messages": [
+                HumanMessage(content="Show me recent DeFi activities for address 0x742d35Cc6634C0532925a3b844Bc454e4438f44e", name="human")]},
+                version="v1")
+
+            async for event in events:
+                if event['event'] == 'on_tool_end':
+                    event_data_input_ = event['data']['input']
+                    self.assertEqual(event['name'], 'DeFiExecutor')
+                    self.assertIn('address', event_data_input_)
```
Comment (naaive) [inline src/tests/agent_trajectory/feed_explore.py:26]: The activity_type is incorrect and should be verified.

image

File: src/openagent/agents/asset_management.py [hunk 2]
```diff
@@ -3,20 +3,22 @@
...
+from openagent.experts.transfer_expert import TransferExpert
 from openagent.tools.nft_balance_tool import NFTBalanceTool
 from openagent.tools.token_balance_tool import TokenBalanceTool
 
 load_dotenv()
 
 asset_management_agent = create_agent(
     get_current_llm(),
-    [TokenBalanceTool(), NFTBalanceTool(), SwapExpert()],
+    [TokenBalanceTool(), NFTBalanceTool(), SwapExpert(), TransferExpert()],
```
Comment (naaive) [inline src/openagent/agents/asset_management.py:14]: add unit test to asset_management

File: src/openagent/experts/defi_expert.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
+from typing import Optional, Type
+
+from langchain.callbacks.manager import AsyncCallbackManagerForToolRun
+from langchain.tools import BaseTool
+from loguru import logger
+from pydantic import BaseModel, Field
+from rss3_dsl_sdk.client import RSS3Client
```
Comment (naaive) [inline src/openagent/experts/defi_expert.py:7]: add poetry dependency

File: src/openagent/app.py
Comment (naaive) [inline src/openagent/app.py:1]: Run `pre-commit run --all-files` to pass the pre-commit lint

Comment (FrankLi123) [reply src/openagent/app.py:1]: completed and passed all cases.

File: src/openagent/experts/defi_expert.py (hunk 3 above)
Comment (FrankLi123) [reply src/openagent/experts/defi_expert.py:7]: used 'poetry add xxx' to add those packages

File: src/openagent/agents/asset_management.py (hunk 2 above)
Comment (FrankLi123) [reply src/openagent/agents/asset_management.py:14]: done.

File: src/tests/agent_trajectory/feed_explore.py (hunk 1 above)
Comment (FrankLi123) [reply src/tests/agent_trajectory/feed_explore.py]: I added an assert check statement here to check the `activity_type`.

Also, I added the logic for the case of `activity_type = all` (while no specific DeFi activity is defined and the DeFiExecutor is triggered, then it searches for all DeFi activities.

**Review by naaive — APPROVED** [review]

**Pull request merged**